src/databricks/sql/utils.py

```diff
@@ -387,6 +388,36 @@ def inject_parameters(operation: str, parameters: Dict[str, str]):
     return operation % parameters
 
 
+def transform_paramstyle(operation: str) -> str:
+    """
+    Performs a regex substitution such that any occurence of `%(param)s` will be replaced with `:param`
+
+    This utility function is built to assist users in the transition between the default paramstyle in
+    this connector prior to version 3.0.0 (`pyformat`) and the new default paramstyle (`named`).
+
+    Args:
+        operation (str): The operation or SQL text to transform.
+
+    Returns:
+        str
+    """
+
+    _output_operation = operation
+
+    PYFORMAT_PARAMSTYLE_REGEX = r"%\((\w+)\)s"
```

Comment:
is this too restrictive?  Can you link to pep or whatever that describes this?  In particular I'm wondering if %( thing )s is allowed.

Reply:
This is the specification: https://peps.python.org/pep-0249/#paramstyle

It's ambiguous as to what format codes are expected to work. But in reality, since we're doing string interpolation under the covers, it doesn't matter so much what PEP-249 requires. Because if customers have queries that depend on this behaviour their queries will break if we don't do something to cover the use case.